OpenTrons/opentrons-api
# feat(protocol-designer): deck v4 touchups and staging area support

closes RAUT-787, RAUT-805, RAUT-806, RAUT-850

# Overview

This PR wires up a few v4 deck migration things and then wires up the staging area slot support (4th column is now selectable on the deck!)

# Test Plan

Make sure your deck modification FF is turned on! 

- create a Flex protocol and select to add a staging area slot
- in the `additional items` section, try to add modules/staging area slots/waste chute and see that things look correct and the slot conflict banner properly shows up 
- move to the deck map and see that the staging area slot renders as expected on the deck map. Hover your mouse over the 4th column where there is a slot and see that the hover works as expected, allowing you to add labware on that slot
- add a labware to the deck
- create a `move labware` form step and move the labware on the deck to the 4th column slot
- now create a `transfer` form step and see that the labware moved to the 4th column is on the aspirate/dispense dropdown. when you create the step, see that the transfer errors because you cannot pipette into a labware on the 4th column
- delete the transfer step and export the protocol
- open the json file and find the `moveLabware` `commandType` command. See that the labware location says that the "slotName" is a column 4 slot
- now import that protocol back into Pd. See that the staging area slot is added to the correct location on the deck map

# Changelog

- various v4 touchups among the edit modules/waste chute/staging area slot components
- fix logic in `getSlotEmpty` util 
- in `deckSetup`, add slot id for labware on top of a module and change type
- fix other utils that rely on slots such as `getUnusedStagingAreaSlot`, `getTrashSlot`, `getSlotEmpty`, `getUnoccupiedSlotOptions`
- extend logic in new location and labware dropdowns to support staging area slots
- remove the `loadFixture` command information from `fileCreator` and the `load_file` reducer and replace with logic to look for the waste chute and staging area slot commands
- create a new `errorCreator` for if a user is trying to pipette into a 4th column slot in step-generation and plug into pipetting atomic commands + add a bit of test coverage

# Review requests

see test plan

# Risk assessment

low
